bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolManager.java

```diff
@@ -105,20 +105,9 @@ protected void afterExecute(Runnable runnable, Throwable throwable) {
             super.afterExecute(runnable, throwable);
 
             if (runnable instanceof Future) {
-                for (Runnable aRunnable : futures.keySet()) {
-                    Future<?> toDelete = null;
-                    synchronized (futures) {
-                        for (Future<?> future : futures.get(aRunnable)) {
-                            if (future == runnable) {
-                                toDelete = future;
-                                break;
-                            }
-                        }
-                        if (toDelete != null) {
-                            logger.trace("Removing Future '{}' (out of {}) for Runnable '{}'", toDelete,
-                                    futures.get(aRunnable).size(), aRunnable);
-                            futures.get(aRunnable).remove(toDelete);
-                        }
+                synchronized (futures) {
+                    for (Runnable aRunnable : futures.keySet()) {
```

Comment:
True for HashMap.  However, 

```java
for (Entry<Runnable, ArrayList<Future<?>>> entry : futures.entrySet()) {
    futures.get(entry.getKey()).removeIf(future -> future == runnable);
}
```

is also less readable. Therefore I thought I leave it as it is.